dir.props

```diff
@@ -2,6 +2,11 @@
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Condition="Exists('..\dir.props')" Project="..\dir.props" />
 
+  <PropertyGroup>
+    <!-- Directory.Build.props imports this .props file, so ensure it isn't imported again -->
+    <ImportDirectoryBuildProps>false</ImportDirectoryBuildProps>
```

Comment:
Consider flipping this property and putting it in `Directory.Build.props` instead.  Like an `<ImportedDirectoryBuildProps>true</ImportedDirectoryBuildProps>`

Reply:
It doesn't work like that.  `ImportDirectoryBuildProps` is a property [declared in MSBuild's `Microsoft.Common.props`](https://github.com/Microsoft/msbuild/blob/38fe952a2ec5885b922e8581d33aa09e0904cf8b/src/Tasks/Microsoft.Common.props#L19).

`Directory.Build.props` imports `dir.props` so it can reuse all the properties in it.  If a non-SDK project imports `dir.props`, and then later imports `Microsoft.Common.props`, it will import `Directory.Build.props`.  We actually don't want `Directory.Build.props` imported at all in the current build for non-SDK projects.

When we take a new CLI, we will be able to detect if we are in an SDK project or not.  Then some more refactoring can happen here.  My thoughts are that we move ALL the properties in `dir.props` into `Directory.Build.props`, and switch the imports around - a blank root `dir.props` that just imports `Directory.Build.props`.